extensions/big-data-cluster/src/wizards/create-cluster/pages/targetClusterPage.ts

```diff
@@ -0,0 +1,173 @@
+		let container = view.modelBuilder.flexContainer().withItems([optionGroup, this.pageContainer], { flex: '0 0 auto' }).withLayout({ flexFlow: 'row', alignItems: 'left' }).component();
+
+		let formBuilder = view.modelBuilder.formContainer().withFormItems(
+			[
+				{
+					component: container,
+					title: ''
+				}
+			],
+			{
+				horizontal: true
+			}
+		).withLayout({ width: '100%', height: '100%' });
+
+		let form = formBuilder.component();
+		await view.initializeModel(form);
+	}
+
+	private createTargetTypeRadioButton(view: sqlops.ModelView, label: string, checked: boolean = false): sqlops.RadioButtonComponent {
+		return view.modelBuilder.radioButton().withProperties({ label: label, name: ClusterTypeRadioButtonGroupName, checked: checked }).component();
+	}
+
+	private initExistingClusterControl(view: sqlops.ModelView) {
+		let self = this;
+		let sectionDescription = view.modelBuilder.text().withProperties({ value: ResourceStrings.ExistingClusterSectionDescription }).component();
+		let configFileLabel = view.modelBuilder.text().withProperties({ value: ResourceStrings.KubeConfigFileLabelText }).component();
+		let configFileInput = view.modelBuilder.inputBox().withProperties({ width: '300px' }).component();
+		let browseFileButton = view.modelBuilder.button().withProperties({ label: ResourceStrings.BrowseText, width: '100px' }).component();
+		let configFileContainer = view.modelBuilder.flexContainer()
+			.withLayout({ flexFlow: 'row', alignItems: 'baseline' })
+			.withItems([configFileLabel, configFileInput, browseFileButton], { CSSStyles: { 'margin-right': '10px' } }).component();
+		this.clusterContextsLabel = view.modelBuilder.text().withProperties({ value: ResourceStrings.ClusterContextsLabelText }).component();
+		this.errorLoadingClustersLabel = view.modelBuilder.text().withProperties({ value: ResourceStrings.ErrorLoadingClustersFromConfigText }).component();
+		this.clusterContextContainer = view.modelBuilder.divContainer().component();
+		this.existingClusterControl = view.modelBuilder.divContainer().withItems([sectionDescription, configFileContainer, this.clusterContextContainer], { CSSStyles: { 'margin-top': '0px' } }).component();
+
+		browseFileButton.onDidClick(async () => {
+			let fileUris = await vscode.window.showOpenDialog(
+				{
+					canSelectFiles: true,
+					canSelectFolders: false,
+					canSelectMany: false,
+					defaultUri: vscode.Uri.file(os.homedir()),
+					openLabel: ResourceStrings.SelectKubeConfigFileText,
+					filters: {
+						'KubeConfig Files': ['kubeconfig'],
+					}
+				}
+			);
+
+			if (!fileUris || fileUris.length === 0) {
+				return;
+			}
+			self.clusterContextContainer.clearItems();
+
+			let fileUri = fileUris[0];
+
+			configFileInput.value = fileUri.fsPath;
+
+			let clusters = self.model.loadClusters(fileUri.fsPath);
+
+			self.cards = [];
+			if (clusters.length !== 0) {
+				self.model.selectedCluster = clusters[0];
+				for (let i = 0; i < clusters.length; i++) {
```

Comment:
Minor: you can do a `self.cards = clusters.map(cluster => ...)` instead of for loop. Just a suggestion